### PR TITLE
[ART-7040] move sync rhcos step to python

### PR DESCRIPTION
**Pull request by Ximinhan**
move sync rhcos srpms step to python.
the sync rhocs srpms step is trying to solve an issue that RHCOS is not using a source from a standard released repo so we need to publish the unreleased srpms.
basically there are two way to handle it:
1 we can no do the sync step and enable this step when this issue happened to make RHCOS correct
2 we can also run this sync by default with filters to prevent this issue happen

this pr is using the 2 method, the filter is set to run the sync against all 4.13+ and not GA releases.

### CODE REVIEW

File: pyartcd/pyartcd/pipelines/promote.py
Comment (joepvd) [inline pyartcd/pyartcd/pipelines/promote.py:376]: name probably should be `sync_rhcos_srpms`

**Review by vfreex — DISMISSED** [review]
/hold
The `config:rhcos-srpms` was a temporary measure to publish sources for unreleased RHCOS content. This is no longer needed at this moment because 4.13, 4.14, and 4.15 are all moved to 9.2 EUS. So this step can be completely removed.

File: pyartcd/pyartcd/pipelines/promote.py
Comment (vfreex) [inline pyartcd/pyartcd/pipelines/promote.py:380]: Currently it is no longer needed for 4.13 and 4.14 since they have moved to 9.2 EUS. In case we need it in the future, maybe we need to turn it into a switch in ocp-build-data.

File: pyartcd/pyartcd/pipelines/promote.py
Comment (vfreex) [inline pyartcd/pyartcd/pipelines/promote.py:395]: For better readability, could you use keyword arguments instead of magical constants `False, False`?

File: pyartcd/pyartcd/pipelines/promote.py
Comment (vfreex) [inline pyartcd/pyartcd/pipelines/promote.py:380]: `assembly.AssemblyTypes.PREVIEW` is also needed. It was missed in the existing code though.

Comment (Ximinhan) [reply pyartcd/pyartcd/pipelines/promote.py]: updated

File: pyartcd/pyartcd/pipelines/promote.py
Comment (joepvd) [inline pyartcd/pyartcd/pipelines/promote.py:384]: Please remove the version logic. Type preview or Candidate is enough.

**Review by joepvd — CHANGES_REQUESTED** [review]
One comment inline. And please rebase your work so the commits make sense without using "the special merge button".

File: pyartcd/pyartcd/pipelines/promote.py
Comment (Ximinhan) [reply pyartcd/pyartcd/pipelines/promote.py]: updated

File: pyartcd/pyartcd/pipelines/promote.py
Comment (vfreex) [inline pyartcd/pyartcd/pipelines/promote.py:383]: I think the decision to is to do this only for `CANDIDATE` and `PREVIEW`.

**Review by joepvd — APPROVED** [review]
/lgtm

**Pull request merged**
